In [1]:
import pandas as pd
import re

In [2]:
dessi_train_labels = pd.read_csv("DeSSI_v2/train_labels.csv")
dessi_train = pd.read_csv("DeSSI_v2/train.csv")

dessi_dev_labels = pd.read_csv("DeSSI_v2/dev_labels.csv")
dessi_dev = pd.read_csv("DeSSI_v2/dev.csv")

dessi_test_labels = pd.read_csv("DeSSI_v2/test_labels.csv")
dessi_test = pd.read_csv("DeSSI_v2/test.csv")

dessi_all = pd.concat([dessi_train, dessi_dev, dessi_test], axis=1)
dataset_split = ["train"] * dessi_train.shape[1] + ["validation"] * dessi_dev.shape[1] + ["test"] * dessi_test.shape[1]
dataset_split_df = pd.DataFrame(dataset_split).T
dataset_split_df.columns = dessi_all.columns
dessi_all = pd.concat([dessi_all, dataset_split_df]).reset_index(drop=True)

dessi_all_labels = pd.concat([dessi_train_labels, dessi_dev_labels, dessi_test_labels], axis=0).reset_index(drop=True)
dataset_split_df = pd.DataFrame(dataset_split)
dataset_split_df.columns = ["split_type"]
dessi_all_labels = pd.concat([dessi_all_labels, dataset_split_df], axis=1).reset_index(drop=True)

C:\Users\Luca\AppData\Local\Temp\ipykernel_9164\3396044622.py:2: DtypeWarning: Columns (62,107,161,241,255,324,449,503,686,720,807,812,889,934,1073,1113,1133,1154,1158,1242,1317,1333,1394,1408,1470,1553,1577,1674,1729,1762,1872,1913,1940,2045,2131,2139,2209,2288,2404,2436,2601,2627,2663,2779,2841,2960,2994,3061,3063,3080,3129,3136,3250,3319,3439,3464,3499,3520,3596,3637,3649,3748,3778,3787,3808,3902,3907,3917,3951,4009,4034,4037,4069,4129,4210,4218,4252,4325,4359,4386,4479,4646,4649,4652,4710,4936,4939,5013,5065,5069,5348,5352,5356,5399,5476,5702,5736,5889,5962,6072,6094,6107,6140,6159,6337,6360,6383,6407,6415,6423,6507,6555,6567,6590,6593,6689,6809,6815,6920,6995,7042,7127,7163,7229,7295,7301,7319,7321,7354,7399,7403,7465,7474,7593,7594,7694,7719,7798,7872,7979,8179,8298,8308,8473,8542,8565,8568,8570,8579,8633,8662,8707,8835,8846,8859,8899,8933,9024,9084,9126,9170,9184,9198,9251,9329,9412,9505,9588,9592,9610,9652,9803,9818,9907,9908,9952,9975,10073,10099,10136,10161,10359,10369,10407,

# Analyze false predictions of CASSED on raw data

In [3]:
predictions = pd.read_csv("../../CASSED/old_predictions/CASSED_dessi/test.tsv",sep="\t")

dessi_labels_cleaned =pd.read_csv("dessi_cleaned/all_labels_cleaned.csv")
wrong_columns = pd.read_csv("dessi_cleaned/wrong_columns.csv")

In [4]:
predictions

email. justin-zavala@banks-jones.us, mareblazincic71@yahoo.com, mia.james3@gmail.com, jordyn.garcia@gmail.com, l.y24@fisher.online, a.salamun252@grguric.com.hr, v-pesut8@milotic.com, beaumcmaster@short-johnson.co, a-dj@zagreb.hr, g.utjesinovic@yahoo.hr, emily.chavez5413@hotmail.com, c.w7590@yahoo.com, alara-nolan7@gmail.com, oliver-bauer@peters-wilson.co, t-gonzalez@mitchell-farmer.biz, magdalena.kardum@glavan.net, rafael.panijan@inet.hr, h-m26@nichols.site, rj@patrick-walker.xyz, kaizer.corbin@moore-wiley.biz, anneliese-thacker@gmail.com, nklipa128@luksa.org, r-robinson179@yahoo.com, esimic4091@kaniski.biz, k.c43@morales-brown.site, e.guest@robinson.ly, i.carrico68@knight.info, cambrieenriquez@hotmail.com, k.h@ross.site, aubrianna.miller@hotmail.com, szivkovic7412@gulan.com.hr, e-b@fitzgerald.online, a-p8824@matesa.biz, vl7852@birkic.net, t.k@zagreb.hr, lthomas@carr-griffin.website, a.storm9846@gmail.com, carl-alonzo1@gmail.com, r-d496@shields.store, davidchambers@andrews.net, leo-givens@adams-blackburn.xyz, h.frazier@gmail.com, patrik-gudelj70@zgorelec.com, a.leyva2639@miller.website, teagan-langley34@carney-bruce.site, c-f@wright-moore.info, l-irvin@richardson-brown.site, crosenberger8059@murphy-graham.website, iriskuckovecki1@zagreb.hr, angietang@watson-jordan.us, j-g@randolph.store, antonio-pilek4@gmail.com, r-burnette@thomas-mckee.info, ellie-dominguez642@hotmail.com, k.lewis147@yahoo.com, faith.ruiz@gaines-walker.org, o.phillips5680@wells-murray.xyz, h.podrug84@karagic.com, d-lee@kennedy-mueller.info, d-m46@xnet.hr, f.li@yahoo.com, m.d@yahoo.com, karias124@hammond-mcconnell.info, j-coleman53@bradshaw-hall.info, leslie-moran@adams-nunez.biz, s.k986@globalnet.hr, a.smith51@walker.us, terezijakezerle9610@zagreb.hr, e-jacobs148@yahoo.com, t.shelton843@klein-knight.us, annabellehaggard68@erickson-kennedy.com, m.danner@armstrong-shelton.online, a.mock9@shelton-mays.website, basler@yahoo.hr, s.k@globalnet.hr, grace.good5491@harper.site, g-a8@gmail.com, leo.adams@anderson.biz, mia.djurdjevic@zagreb.hr, a-barlow623@hotmail.com, v.blazevic9@net.hr, kstewart@reynolds.org, sara-eldic@filipovic.info, s-v@matthews-english.net, alexaescoto90@lara-thomas.co, vlado.kolar@zagreb.hr, n.d@globalnet.hr, ms514@yahoo.com, gl66@globalnet.hr, olivia.nesbitt@hotmail.com, d.thornton39@yahoo.com, milan.bicanic9036@matijas.info, g-day@olsen.org, m-winter6062@beasley-schroeder.org, g.v@zg.t-com.hr, a-t@hotmail.com, c-west@bradley-flores.site, d.parker7@osborn.ly, mperic6328@t.ht.hr, osaiz9@hotmail.com
0                                  - Gold: [Email (1.0)]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            

In [5]:
def extract_prediction(i):
    list_extractor = re.compile(r"(\w+)\s*\(\d+(\.\d+)?\)")
    matches = list_extractor.findall(predictions.iloc[i, :].values[0])
    return [match[0] for match in matches]

def convert_predictions(predictions):
    column_names = []
    gold_list = []
    pred_list = []
    mismatch = []
    column_names.append(predictions.columns[0].split(". ")[0].strip())

    height = predictions.shape[0]

    for i in range(height):
        if 'Gold' in predictions.iloc[i, :].values[0][:10]:
            gold_list.append(extract_prediction(i))
        elif 'Pred' in predictions.iloc[i, :].values[0][:10]:
            pred_list.append(extract_prediction(i))
            if i == height-1:
                mismatch.append("No Mismatch")
                break
            if 'MISMATCH' in predictions.iloc[i+1, :].values[0][:20]:
                mismatch.append("Mismatch")
            else:
                mismatch.append("No Mismatch")
        elif 'MISMATCH' not in predictions.iloc[i, :].values[0][:20]:
            column_names.append(predictions.iloc[i, :].values[0].split('. ')[0].strip())
            


    df = pd.DataFrame({
        'Name': column_names,
        'Gold': gold_list,
        'Pred': pred_list,
        'Mismatch': mismatch
    })
    return df

In [6]:
df = convert_predictions(predictions)
df

Name                    Gold                    Pred  \
0                     email                 [Email]                 [Email]   
1              ldfncafwhfzv          [Phone_number]          [Phone_number]   
2                 email_sid                 [Email]                 [Email]   
3               phonenumber          [Phone_number]          [Phone_number]   
4              pxsgydcjmcmj                [Person]                [Person]   
...                     ...                     ...                     ...   
6267              latdeg.12           [Geolocation]           [Geolocation]   
6268             genderid.7            [Other_data]            [Other_data]   
6269           ztyhgaddzlun                   [NIN]                   [NIN]   
6270        geo_latitude.18  [Geolocation, Address]  [Geolocation, Address]   
6271  date_account_closed.6                  [Date]                  [Date]   

         Mismatch  
0     No Mismatch  
1     No Mismatch  
2     No Mismatch  
3     No Mismatch  
4     No Mismatch  
...           ...  
6267  No Mismatch  
6268  No Mismatch  
6269  No Mismatch  
6270  No Mismatch  
6271  No Mismatch  

[6272 rows x 4 columns]

In [7]:
df.Mismatch.value_counts()

No Mismatch    6207
Mismatch         65
Name: Mismatch, dtype: int64

In [8]:
#add labels and predictions to dessi dataset

labels_transposed =  dessi_test_labels.T
labels_transposed.columns = dessi_test.columns
dessi_test = pd.concat([dessi_test, labels_transposed])
predicted_transposed = pd.DataFrame(df.Pred).T
predicted_transposed.columns = dessi_test.columns
dessi_test = pd.concat([dessi_test, predicted_transposed])

Check which Mismatches could have resulted of labeling errors

In [9]:
pd.set_option('display.max_columns', 70)
dessi_test[df[df.Mismatch == "Mismatch"].Name.values].iloc[[0,1,2,-2,-1],:]

gsrvmkhhhsjn                      useremail.1   invoice_details  \
0         +55 71 8072 4850                   e-e@russell.ly              Serb   
1               1296774986  athena.lacy8@hill-james.website  Country of absce   
2             095 6189 104              b.light27@yahoo.com  Country of absce   
label         Phone_number                     Email,Person        Other_data   
Pred   [Phone_number, NIN]                          [Email]                []   

            dtyzjfihtxvj nccvantofrjh         janorvodayhw  zdittbhttzwu  \
0               SLWXGBJF  497-61-6103           4312267413      UELJGB6U   
1            HQCDGB9B9H9  891-52-5150           4310638148      SLRHGBS8   
2               FBYRGBME  225-99-4363           4319300834      JXSZGBP4   
label          SWIFT/BIC          NIN         Phone_number     SWIFT/BIC   
Pred   [Other_data, BIC]  [Date, NIN]  [Phone_number, CCN]  [Other_data]   

                  fax.1       zyvgmmultuuz         jnmbgxurxnmj  xdjgpfjzidee  \
0           047 383 777        QFBQGB8MLA3    002-504-3954x1993   IZCJGBCBN9U   
1            2576458199        XPZCGBY01CT    190-568-4989x7242   EWVXGBW7MZK   
2            2575104670        PCIOGBOALGR    154-977-9024x7909   QGRQGBFQ8R8   
label  Phone_number,NIN         Other_data         Phone_number     SWIFT/BIC   
Pred     [Phone_number]  [Other_data, BIC]  [Phone_number, NIN]  [Other_data]   

       gfgxyrajzwtb             brtatrqkunkx         uukxetrnhxck  \
0         HK6103647     Jairus Seth Knowlton             H2362048   
1         ZS7670597  Riaan Angelique Meneses             Z2929190   
2         ZM5654042                    Margo             M3608374   
label      Passport             Email,Person             Passport   
Pred   [Other_data]                 [Person]  [Passport, ID_Card]   

          family_contact.6   contactperson.8 wtpkwunxmsve  \
0      +52 (041) 6901 0308        2530494216     P3278680   
1              580 143 857       05186739245     M3744966   
2      '+98 (2) 052-176575       76318092549     D5714224   
label     NIN,Phone_number  NIN,Phone_number      ID_Card   
Pred        [Phone_number]    [Phone_number]   [Passport]   

                        insee_code.2       p_id  wbwdbqtngjbx lcrcjsagqjvv  \
0                        285 014 437   M2960577   19254654565  760 324 558   
1      a-b9791@adams-townsend.online   Q7350505    6799138826  615 275 344   
2             y-j4@decker-nunez.info   U9932315   12860299457  556 108 348   
label                      Email,NIN   Passport  Phone_number          NIN   
Pred           [Phone_number, Email]  [ID_Card]         [NIN]  [Date, NIN]   

              hzxgmsuvjjav                          first_name.9  \
0                270734431  out of the box  - events productions   
1                276131641               Polygon fire protection   
2                279367956                        Hill Top Farms   
label         Phone_number                   Person,Organization   
Pred   [Phone_number, NIN]                        [Organization]   

       gbldfrjrtrjd wpuawkihjvgg          hevivdvucvbp  fsvvxcsbaozi  \
0          QHHVGBU0     W3924781                Female   892  410426   
1       OWKFGB2COIH     X8632888                Female   117  071791   
2       KREOGBRKRX4     G4652166                  Male   399  950899   
label     SWIFT/BIC     Passport                Gender  Phone_number   
Pred   [Other_data]    [ID_Card]  [Other_data, Gender]         [NIN]   

             managersid.9       utulyzhmdgqk lfjbqqpqzewq      nsmetka3  \
0              MKCC193950        SUPZGBTFIVH  NUEOGBL2HG0      KVYKGBVO   
1              MKCC194494           FLLJGBAM  OYZSGB2ARF2      JQFMGBL6   
2              MKCC488791        CTALGBS3LO8     RTQLGBOV      EWLWGB2F   
label          Other_data         Other_data   Other_data     SWIFT/BIC   
Pred   [Other_data, IBAN]  [Other_data, BIC]        [BIC]  [Other_data]   

      family_contact_email.24      phon

In [10]:
mismatches = []
for i in (dessi_test[df[df.Mismatch == "Mismatch"].Name.values].iloc[[-2, -1],:].T.values):
    mismatches.append(i)
pd.Series(mismatches).value_counts()

[Phone_number, [Phone_number, NIN]]    1
[Person,Email, [Email]]                1
[Phone_number, [Phone_number, NIN]]    1
[Other_data, [BIC]]                    1
[NIN,Email, [Email]]                   1
                                      ..
[Other_data, [Other_data, IBAN]]       1
[Other_data, [Other_data, BIC]]        1
[Other_data, [BIC]]                    1
[SWIFT/BIC, [Other_data]]              1
[Address, [Address, GPE]]              1
Length: 65, dtype: int64

In [11]:
(dessi_test["emailladdresssid.16"].values)

array(['craig.roberson621@price-harris.biz', 'r.m6@yahoo.com',
       'ah@gmail.com', 'd.a30@sullivan-lee.store', 'khorvat@gmail.com',
       'leon-petrekovic2406@kupsjak.hr', 'steven-jackson77@mcdonald.info',
       'payton.jackson4@irwin-stone.store', 'leni-glavas@zg.t-com.hr',
       'nyomicasey@fisher-frey.info', 'mh586@gmail.com',
       'golson9777@wiggins-fleming.store', 'c-b@spears-stokes.org',
       'charles-flores2020@roberts-butler.store',
       'lilypotts221@henson-lee.site', 'k.lofton95@vincent.online',
       'b-baucom@joyce-roy.info', 'j.m86@yahoo.com',
       'mhoffman@cervantes.us', 'Mihajlović',
       'krosales964@crawford-roberts.ly',
       'o-gutierrez@diaz-simmons.website', 'Jackson',
       'theodore-aguilar@stewart-cline.website',
       'rosamiles88@king-ruiz.xyz', 'e.rangel4041@garza.com',
       'chloe-lehman7@herring-frederick.website', 'Simon',
       'olivia.gil501@davis-wheeler.co', 'd.d68@long.biz',
       'k.j262@larsen-vasquez.us', 'b.shafer@garcia-

useremail.1 does not contain Person information  
brtatrqkunkx, does not contain emails   
family_contact_email.24 does not contain emails, it is only the column name    
customer_lname.5, only contains Emails    
resident_registration_number.10, does not contain NIN  
ssnid.5, does not contain emails  
fiscal_code.7, does not contain person information    



Phone_number, NIN difficult to validate, often it seems to be only Phone_number, this combination leads to most faults, they often have conflicts which is understandable  
BIC also causes multiple mistakes  

In [12]:
wrong_columns = list(wrong_columns["0"].values)

In [13]:
columns = ["brtatrqkunkx", "fiscal_code.7", "useremail.1", "customer_lname.5", "family_contact_email.24", "resident_registration_number.10", "ssnid.5"]
indis = []
for e, i in enumerate(dessi_all.columns):
    if (i in columns) and ("test" in dessi_all.iloc[100,e]) and (e not in wrong_columns):
        indis.append(e)
        print(i)
len(indis)

useremail.1
customer_lname.5
resident_registration_number.10
fiscal_code.7


4

In [14]:
for ind in indis:
    wrong_columns.append(ind)
dessi_labels_cleaned.loc[indis]

label split_type
25458  Email,Person       test
29078  Person,Email       test
29324     NIN,Email       test
30198    NIN,Person       test

In [15]:
dessi_labels_cleaned.loc[indis[0], "label"] = "Email"
dessi_labels_cleaned.loc[indis[1],  "label"] = "Email"
dessi_labels_cleaned.loc[indis[2],  "label"] = "Email"
dessi_labels_cleaned.loc[indis[3],  "label"] = "NIN"
dessi_labels_cleaned.loc[indis]

label split_type
25458  Email       test
29078  Email       test
29324  Email       test
30198    NIN       test

For some columns the column name fit to the labels, but they contain no value fitting this label, e.g. customer_lname.5

# Check column names

Another hint that CASSED is overfitting of dessi is the appeareance of column names fitting to a class but having a "Other_data" label (e.g. gender) 

In [16]:
indis = []
for i in range(6272):
    if set(dessi_test.iloc[:100, i].values) == {"M", "F", "NB"}:
        indis.append(i)
dessi_test.iloc[:,indis]

zbpozurvpbcj  fqhezsttzrgd      sex_id.2         sex.2      gender.2  \
0                NB             M             M            NB            NB   
1                NB             M             M             F             F   
2                 F             M             M            NB            NB   
3                NB             F             M             F            NB   
4                NB             F            NB            NB             F   
...             ...           ...           ...           ...           ...   
97                F             M             M             F             M   
98                F             M             F             M             M   
99               NB             F             F             F            NB   
label    Other_data    Other_data    Other_data    Other_data    Other_data   
Pred   [Other_data]  [Other_data]  [Other_data]  [Other_data]  [Other_data]   

         genderid.6         sex.8  
0                 M            NB  
1                 F             M  
2                NB             M  
3                NB             F  
4                NB            NB  
...             ...           ...  
97               NB             M  
98                M             M  
99                F             F  
label    Other_data    Other_data  
Pred   [Other_data]  [Other_data]  

[102 rows x 7 columns]

The columns with sex or gender in the column name have to be relabeld

In [17]:
columns = dessi_test.iloc[:,indis].columns
indis = []
for e, i in enumerate(dessi_all.columns):
    if (i in columns) and ("test" in dessi_all.iloc[100,e]):
        indis.append(e)
for ind in indis:
    wrong_columns.append(ind)
dessi_labels_cleaned.loc[indis,  "label"] = "Gender"

In [18]:
labels_transposed =  dessi_all_labels.T
labels_transposed.columns = dessi_all.columns
dessi_all = pd.concat([dessi_all, labels_transposed])
dessi_all = dessi_all.reset_index(drop=True)

In [19]:
print(dessi_all.iloc[101,:].value_counts().get("Gender"))
indis = []
for i in range(dessi_all.shape[1]):
    if dessi_all.iloc[101,i] == "Gender":
        indis.append(i)
print(len(indis))
dessi_all.iloc[:,indis]

94
94


gen type         pol              kgebmormtjgy   genderclass  \
0       Male  Neodređeno                    Hijras      Demienby   
1     Female  Neodređeno       Masculine-of-center     trans men   
2       Male  Neodređeno                      MAAB          MAAB   
3       Male  Neodređeno                      MAAB      Bigender   
4       Male       Muško                   Agender             F   
..       ...         ...                       ...           ...   
98      Male      Žensko  Female Assigned At Birth  Third gender   
99    Female      Žensko                Polygender    Novigender   
100    train       train                     train         train   
101   Gender      Gender                    Gender        Gender   
102    train       train                     train         train   

              gender_specific mqmymhjoqjuh gender identity gender related  \
0                   Trigender        Kadın               F         Female   
1          Feminine-of-center        Erkek               M         Female   
2                Third gender        Kadın               F           Male   
3              Male-to-female        Kadın               F           Male   
4    Female Assigned At Birth        Erkek               M           Male   
..                        ...          ...             ...            ...   
98                   Demienby        Kadın               M           Male   
99                      Hijra        Erkek               F         Female   
100                     train        train           train          train   
101                    Gender       Gender          Gender         Gender   
102                     train        train           train          train   

                 ljbqclscqgxi    gender.1 gender classification   pol.1  \
0                   Cis woman       Muško                 Moški       M   
1                 Transgender       Muško                Ženski       M   
2                   Demitrans  Neodređeno                Ženski       M   
3                  Non binary       Muško                Ženski       M   
4                  Novigender       Muško                Ženski       M   
..                        ...         ...                   ...     ...   
98   Female Assigned At Birth      Žensko                Ženski       M   
99                Stone butch      Žensko                 Moški       M   
100                     train       train                 train   train   
101                    Gender      Gender                Gender  Gender   
102                     train       train                 train   train   

                      gndr_id ntqttkvhbjih tip na rod####12140743  \
0               Non cisgender            Ž               Arsenikó   
1                    Demienby            M                Thilykó   
2                  Non binary            M               Arsenikó   
3                  Non binary            Ž                Thilykó   
4    Assigned Female At Birth            Ž                Thilykó   
..                        ...          ...                    ...   
98     Male Assigned At Birth            M                Thilykó   
99                        Boi            Ž               Arsenikó   
100                     train        train                  train   
101                    Gender       Gender                 Gender   
102                     train        train                  train   

             genderidentity gndr id gender specific gen type.1  \
0                       MTF       M        Arsenikó      Kadın   
1                      FAAB       M         Thilykó      Kadın   
2            Transmasculine       F        Arsenikó      Erkek   
3                Non binary       F        Arsenikó      Kadın   
4       Masculine-of-center       M        Arsenikó      Erkek   
..                      ...     ...             ...        ...   
98                Pangender       M        Arsenikó      Kadın   
99  

These are recognized genders in different languages

In [20]:
indis = []
pd.set_option("display.max_columns", 500)
for i in range(dessi_all.shape[1]):
    if ("gender" in dessi_all.columns[i].lower() or "sex" in dessi_all.columns[i].lower()) and (dessi_all.iloc[101,i] != "Gender"
                                                                                                  and dessi_all.iloc[101,i] != "Sexuality"):
        # two columns which contain sex but are not sex related
        if dessi_all.columns[i] in ["asexklxtuarm", "pzjrssexpfve", "qqfmkosexymj", "ebcacsexfrfn", "uwpejzvksexb", "hsexjtkdydff", 
                                    "ahsexkqmvqhy", "lzvfbbosexyt", "sexlvfrjndwe"]:
            continue
        indis.append(i)
df_gender = dessi_all.iloc[:,indis]
df_gender

sex       sex.1      sex_id    sex_id.1       sex.2      gender  \
0            no           0          NB           1           F           1   
1           yes           0          NB           1           F           1   
2            no           1           F           1           M           1   
3           yes           1           M           0           M           1   
4           yes           0           M           1           M           1   
..          ...         ...         ...         ...         ...         ...   
98           no           0           M           0           M           1   
99          yes           1           F           1           M          -1   
100       train       train       train       train       train       train   
101  Other_data  Other_data  Other_data  Other_data  Other_data  Other_data   
102       train       train       train       train       train       train   

          sex.3   gender_id    gender.2    sex_id.2    gender.3    sex_id.3  \
0          Male          no          no        Male         boy         boy   
1        Female         yes         yes      Female         boy         boy   
2        Female         yes          no      Female        girl         boy   
3          Male          no          no        Male        girl         boy   
4          Male         yes         yes        Male        girl         boy   
..          ...         ...         ...         ...         ...         ...   
98       Female         yes          no      Female         boy         boy   
99         Male          no         yes        Male        girl         boy   
100       train       train       train       train       train       train   
101  Other_data  Other_data  Other_data  Other_data  Other_data  Other_data   
102       train       train       train       train       train       train   

       gender.4    sex_id.4       sex.4    gender.5       sexid    genderid  \
0           yes           1           M           F           F          no   
1            no           0           F           M           M         yes   
2            no           1           M           F           M          no   
3           yes           0           M           F           F          no   
4           yes           1           M           M           F         yes   
..          ...         ...         ...         ...         ...         ...   
98          yes           1           M           M           F         yes   
99           no           1           M           F           F          no   
100       train       train       train       train       train       train   
101  Other_data  Other_data  Other_data  Other_data  Other_data  Other_data   
102       train       train       train       train       train       train   

     genderid.1 gender_id.1  genderid.2    gender.6    sex_id.5  genderid.3  \
0          Male        girl           M           F         boy        girl   
1        Female         boy          NB           M         boy        girl   
2          Male        girl          NB           M        girl        girl   
3        Female         boy           M           F        girl        girl   
4          Male        girl           M           F         boy         boy   
..          ...         ...         ...         ...         ...         ...   
98         Male         boy          NB           M         boy         boy   
99         Male        girl           M           M         boy        girl   
100       train       train       train       train       train       train   
101  Other_data  Other_data  Other_data  Other_data  Other_data  Other_data   
102       train       train       train       train       train       train   

       sex_id.6  genderid.4    sex_id.7    sex_id.8  genderid.5    gender.7  \
0             M          no        Male        Male         yes           M   
1            NB         yes      Female        Male       

143 columns have the real label gender but are all wrong labeled

In [21]:
indis = []
for e, (i, t) in enumerate(zip(dessi_all.columns, dessi_all.iloc[100,:])):
    try:
        if (i in df_gender.columns) and (t in df_gender.loc[100,i].values):
            indis.append(e)
    except:
        if (i in df_gender.columns) and (t in df_gender.loc[100,i]):
            indis.append(e)


for ind in indis:
    wrong_columns.append(ind)
dessi_labels_cleaned.loc[indis, "label"] = "Gender"

Search for female/male/girl/boy values in columns

In [22]:
keywords = ["Male", "Female", "girl", "boy", "male", "female", "Girl", "Boy", "MALE", "FEMALE", "GIRL", "BOY"]
indis = []
for e,i in enumerate(dessi_all.columns):
    some_vals = dessi_all[i].values[0:5]
    if e not in wrong_columns and any(keyword in some_vals for keyword in keywords):
        indis.append(e)
dessi_all.iloc[:,indis]

gbhwunppuetx kkutwrzyqmep gen type gender related           pol.2  \
0           Male         girl     Male         Female  Gender diverse   
1         Female          boy   Female         Female  Gender diverse   
2         Female         girl     Male           Male          Female   
3         Female          boy     Male           Male            Male   
4           Male         girl     Male           Male          Female   
..           ...          ...      ...            ...             ...   
98          Male         girl     Male           Male          Female   
99          Male         girl   Female         Female          Female   
100        train        train    train          train           train   
101   Other_data   Other_data   Gender         Gender          Gender   
102        train        train    train          train           train   

    xhibvzhdmhkr gdjjezmvrpaz gender related.2 g classification aknbfvthxzyi  \
0         Female          boy           Female           Ženski          boy   
1         Female         girl   Gender diverse            Moški          boy   
2         Female         girl             Male           Ženski         girl   
3           Male         girl             Male           Ženski          boy   
4         Female         girl           Female            Moški         girl   
..           ...          ...              ...              ...          ...   
98        Female          boy           Female           Ženski          boy   
99          Male          boy           Female           Ženski          boy   
100        train        train            train            train        train   
101       Gender   Other_data           Gender           Gender   Other_data   
102        train        train            train            train        train   

    rodova identitet gndr class      gndr_type gender classification.1  \
0              Femer     Female   Multi-gender                    Male   
1              Femer       Male  Transfeminine                  Female   
2           Mashkull     Female     Demifemale                  Female   
3              Femer       Male         Female                  Female   
4           Mashkull       Male          Trans                  Female   
..               ...        ...            ...                     ...   
98          Mashkull     Female    Transsexual                  Female   
99             Femer     Female          Butch                    Male   
100            train      train          train                   train   
101           Gender     Gender         Gender                  Gender   
102            train      train          train                   train   

    gndr id.3 qnfcstrldrmx              qiaikfguwhgg zjfiirysavti  \
0        Male         Male    Assigned Male At Birth         girl   
1      Female       Female                      Male         girl   
2      Female         Male            Transmasculine         girl   
3      Female         Male                     Hijra          boy   
4      Female         Male                 Nonbinary          boy   
..        ...          ...                       ...          ...   
98     Female       Female  Designated Male At Birth          boy   
99       Male       Female    Male Assigned At Birth         girl   
100     train        train                     train        train   
101    Gender       Gender                    Gender   Other_data   
102     train        train                     train        train   

       gender class qdcevpnnhyaz rinvlrtrbyah xbsweikildqj gender class  \
0    Gender diverse       Female         Male       Female         Male   
1            Female       Female         Male         Male       Female   
2    Gender diverse         Male         Male       Female       Female   
3              Male       Female         Male         Male       Female   
4              Male       Female       Female         Male         Male  

In [23]:
counter = 0
indis_2 = []
for ind in indis:
    if (dessi_labels_cleaned.loc[ind, "label"] != "Gender") & (dessi_labels_cleaned.loc[ind, "label"] != "Sexuality"):
        counter +=1
        indis_2.append(ind)
for ind in indis_2:
    wrong_columns.append(ind)
dessi_labels_cleaned.loc[indis, "label"] = "Gender"

Differences between classes like Geolocation and address (when only the city is given is questionable),   
CASSED always gets the right label even if the difference can't be examined (f1-score)   
Same goes for the classes NIN, ID_Card and Passport

# Label Errors that have been found during testing of dessi data on CASSED

In [24]:
for i in range(dessi_all.shape[1]):
    if dessi_all.columns[i] == "sex.3" and dessi_all.iloc[-1, i] == "test":
        print(i)

29355


In [25]:
dessi_all.iloc[:, 29355]

0            Akoisexual
1         Abrosexuality
2      Nonheterosexuals
3           Monosexuals
4         Novosexuality
             ...       
98         Sapiosexuals
99         Lithosexuals
100                test
101           Sexuality
102                test
Name: sex.3, Length: 103, dtype: object

In [26]:
dessi_labels_cleaned.iloc[29355, 0]

'Gender'

In [27]:
dessi_labels_cleaned.iloc[29355, 0] = "Sexuality"
wrong_columns.append(29355)
pd.DataFrame(wrong_columns).to_csv("dessi_cleaned/wrong_columns.csv", index=False)
dessi_labels_cleaned.to_csv("dessi_cleaned/all_labels_cleaned.csv", index=False)

In [28]:
pd.DataFrame(dessi_labels_cleaned.loc[:18815,"label"]).to_csv("dessi_cleaned/train_labels.csv", index=False)
pd.DataFrame(dessi_labels_cleaned.loc[18816:18816+6271,"label"]).to_csv("dessi_cleaned/dev_labels.csv", index=False)
pd.DataFrame(dessi_labels_cleaned.loc[18816+6272:,"label"]).to_csv("dessi_cleaned/test_labels.csv", index=False)
print(f"Currently {len(wrong_columns) / dessi_all.shape[1] * 100}% of the columns of the test set are wrong labeled.")

Currently 2.541454081632653% of the columns of the test set are wrong labeled.
